src/labone/mock/hpk_functionality.py

```diff
@@ -0,0 +1,59 @@
+from abc import ABC
+from contextlib import contextmanager
+import json
+
+from labone.core import ListNodesFlags, ListNodesInfoFlags
+from labone.core.helper import LabOneNodePath
+from labone.core.session import NodeInfo
+from labone.core.subscription import StreamingHandle
+from labone.core.value import AnnotatedValue
+from labone.mock.mock_server import ConcreteMockServer
+
+
+class HpkMockFunctionality(ABC):
+    async def get(self, path: LabOneNodePath) -> AnnotatedValue:
+        raise NotImplementedError
```

Comment:
question: And If the class is an abstract class these methods should be abstract as well ... The deriving class can throw a NotImplementedError if they like but not the abstract class

Reply:
I considered this. The downside is, that with abstract methods, your are forced to override all of them in every implementation. For mock servers not relying on all methods, this is an extra burden. Therefore I find the NotImplementedError preferable

Reply:
In my opinion, we should ensure that all the methods are overwritten. The client relies on the fact that all interface functions are implemented ... if not, it will end up in weird exceptions stating that the version is wrong and maybe even close the connection.

If a user does not want to overwrite all methods, than he should use a higher level abstraction